server/schedule/operator_controller.go

```diff
@@ -147,7 +147,21 @@ func (oc *OperatorController) Dispatch(region *core.RegionInfo, source string) {
 	}
 }
 
-func (oc *OperatorController) checkStaleOperator(op *operator.Operator, region *core.RegionInfo) bool {
+func (oc *OperatorController) checkStaleOperator(op *operator.Operator, step operator.OpStep, region *core.RegionInfo) bool {
+	switch s := step.(type) {
+	case operator.RemovePeer:
+		if s.FromStore == region.GetLeader().GetStoreId() {
```

Comment:
I think it's too tricky to make special judgments on removePeer and leader here. Maybe we can solve it by recording the expected region leader at each step and check if the leader is the same. (like `op.ConfVerChanged`)